pnpm-workspace.yaml

```
@@ -7,3 +7,4 @@ packages:
   - "packages/languages"
   - "packages/tasks"
   - "packages/widgets"
+  - "packages/templates"
```

```suggestion
  - "packages/chat_templates"
```

`chat_templates` (or `chat-templates`?) would be clearer no?

---

Sure, I can change it 😇 I originally opted for just "templates" because the library can actually also be used outside of an ML context, but with our focus being quite clear (i.e., minimalistic and specifically designed for ChatML formats), it would probably make more sense to rename it.

---

if it's just going to be the current scope, you can also name it `@huggingface/jinja` i guess 

Depends on the planned scope!

---

discussing more with @gary149 and others i kinda like `@huggingface/jinja` too – do you think this package will include (or already does include) more stuff than just a jinja parser?

---

Yep, it's just jinja at the moment, and it probably won't change unless we end up switching this in the python library. In fact, the library in its current state doesn't reference or use chat-templating-specific logic. I don't mind calling it `@huggingface/jinja`, but it should be clear to users what they are getting: something which is not a truly jinja-compatible templating library. The library is very limited in scope, and cannot (and probably shouldn't) support all functionality of a full templating engine.

For users who do want a feature-complete jinja-like templating engine, they can use [nunjucks](https://github.com/mozilla/nunjucks), however, that is not suitable for our needs since (as stated in [getting started guide](https://mozilla.github.io/nunjucks/getting-started.html)):

> nunjucks does not sandbox execution so it is not safe to run user-defined templates or inject user-defined content into template definitions. On the server, you can expose attack vectors for accessing sensitive data and remote code execution. On the client, you can expose cross-site scripting vulnerabilities even for precompiled templates (which can be mitigated with a strong [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy)). See [this issue](https://github.com/mozilla/nunjucks-docs/issues/17) for more information.

And all our templates are user-defined. 😅 nunjucks works by converting the template string to javascript code, then evaluating it as javascript code (see [this line of code](https://github.com/mozilla/nunjucks/blob/ea0d6d5396d39d9eed1b864febb36fbeca908f23/nunjucks/src/environment.js#L565)). In contrast, this library never evaluates anything with `eval`, `new Function`, or anything else like it. Instead, we create our [own runtime](https://github.com/huggingface/huggingface.js/blob/277163baf450bacf0a96750a5fe2a7d4269da7f0/packages/chat-templates/src/jinja/runtime.ts), and only expose very specific functionality to the user. See [here](https://github.com/huggingface/huggingface.js/blob/277163baf450bacf0a96750a5fe2a7d4269da7f0/packages/chat-templates/src/jinja/runtime.ts#L58-L77) for example, to see how we handle instance methods.

Naming it `@huggingface/jinja` will definitely help with SEO though 😆

**TLDR**: We could, but it might be misleading to users.

---

i'd call it `@huggingface/jinja` then if there's nothing specific to chat-templates. But i'll let you decide!